llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/IR/LinalgExtOps.cpp

```diff
@@ -1770,6 +1777,31 @@ inferPackedType(ShapedType sourceType, ArrayRef<int64_t> innerTiles,
 // PackOp
 //===----------------------------------------------------------------------===//
 
+/// Custom builder methods for pack ops.
+void PackOp::build(OpBuilder &builder, OperationState &state, Value source,
+                   Value output, ArrayRef<int64_t> innerDimsPos,
+                   ArrayRef<OpFoldResult> innerTiles,
+                   Optional<Value> paddingValue,
+                   ArrayRef<int64_t> outerDimsPerm) {
+  assert(innerDimsPos.size() == innerTiles.size());
+  DenseMap<int64_t, OpFoldResult> tileAndPosMapping;
+  for (auto it : llvm::zip(innerDimsPos, innerTiles))
+    tileAndPosMapping[std::get<0>(it)] = std::get<1>(it);
+  SmallVector<int64_t> staticTileSizes;
+  SmallVector<Value> dynamicTileSizes;
```

Comment:
Should we provide an helper to get `dynamicTileSizes` directly in the op? I am mentioning this because we already have an helper for `staticTileSizes` (aka: `getStaticTiles`).

Reply:
I am not sure what that would look like. I cant see a use case right now outside of this `build` method.